docs/overrides/javascript/client.js

```
@@ -0,0 +1,900 @@
+    const timeEl = this.qs(".ult-thinking-time", thinking);
+    const t0 = performance.now();
+    const tick = setInterval(() => {
+      timeEl.textContent = `(${((performance.now() - t0) / 1000).toFixed(1)}s)`;
+    }, 100);
+    this.scrollToBottom();
+
+    this.abortController = new AbortController();
+
+    try {
+      const res = await fetch(this.apiUrl, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          messages: [{ role: "user", content: text }],
+          session_id: this.sessionId,
+        }),
+        signal: this.abortController.signal,
+      });
+      if (!res.ok) throw new Error(`HTTP ${res.status}`);
+
+      const sid = res.headers.get("X-Session-ID");
+      if (sid && !this.sessionId) {
+        this.sessionId = sid;
+        this.saveSessionId(sid);
+      }
+
+      thinking.remove();
+      clearInterval(tick);
+
+      const div = this.el("div", "ult-message assistant");
+      group.appendChild(div);
+
+      const reader = res.body.getReader();
+      const decoder = new TextDecoder();
+      let content = "";
+
+      while (true) {
+        const { done, value } = await reader.read();
+        if (done) break;
+        const chunk = decoder.decode(value, { stream: true });
+        for (const line of chunk.split("\n")) {
+          if (!line.startsWith("data: ")) continue;
+          const data = line.slice(6);
+          if (data === "[DONE]") continue;
+          try {
+            const parsed = JSON.parse(data);
+            if (parsed.content) {
+              content += parsed.content;
+              div.innerHTML = this.renderMarkdown(content);
+              this.scrollToBottom();
+            } else if (parsed.error) {
+              throw new Error(parsed.error);
+            }
+          } catch (e) {
+            if (e.message !== "Unexpected end of JSON input")
+              console.error("Parse error:", e);
+          }
+        }
+      }
```

❗ **CRITICAL**: `chunk.split("\n")` assumes each read delivers whole SSE lines. In practice the stream can split inside a JSON payload, so the first fragment fails `startsWith("data: ")` and the subsequent fragment keeps the remainder without the `data: ` prefix. That drops tokens and the assistant response silently truncates. Please buffer unfinished lines between reads (e.g. keep a `pending` string outside the loop, prepend it to the next chunk, and only parse when you have a full line) before calling `JSON.parse`.